server-common/src/main/java/org/apache/kafka/server/share/persister/PersisterStateManager.java

```diff
@@ -237,6 +237,11 @@ public PersisterStateManagerHandler(
             this.onCompleteCallback = response -> {
             }; // noop
             partitionKey = SharePartitionKey.getInstance(groupId, topicId, partition);
+            String canonicalName = getClass().getCanonicalName();
+            if (canonicalName == null) {
+                canonicalName = getClass().getName();
+            }
+            log = LoggerFactory.getLogger(canonicalName);
```

Comment:
nit: probaly can do later as well. Becuase now it's not always the canonical name in the variable.

```suggestion
            String name = getClass().getCanonicalName();
            if (name == null) {
                name = getClass().getName();
            }
            log = LoggerFactory.getLogger(name);
```